internal/service/resource_object_mapper.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2023 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+compliance with the License. You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is
+distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing permissions and limitations under the
+License.
+*/
+
+package service
+
+import (
+	"context"
+
+	"github.com/openshift-kni/oran-o2ims/internal/data"
+)
+
+//go:generate mockgen -source=resource_object_mapper.go -package=service -destination=resource_object_mapper_mock.go
+type ResourceObjectMapper interface {
```

Comment:
There is already a `Mapper` type here:

https://github.com/openshift-kni/oran-o2ims/blob/ccd39ba2f1c2f016566be75721cc824d6d8870c5/internal/streaming/stream.go#L145-L146

It is a simple function, so I think we can use that directly instead of creating a new interface and a mock for it.

If using the generic type is unconfortable we can add an alias in the `data` package:

```go
type Mapper = stream.Mapper[data.Object, data.Object]
```